File: pset6/credit/credit.py
from cs50 import get_int

while True:
    number = get_int("Number:")
    if number >= 0:
        break
sum_digits = 0
sum_odd = 0
card_type = 0
while number > 0:
    if number > 9 and number < 100:
        card_type = number
    sum_odd = sum_odd + number % 10
    number = (number - number % 10) / 10
    digit = (number % 10) * 2
    if number > 9 and number < 100:
        card_type = number
    while digit > 0:
        sum_digits = sum_digits + (digit % 10)
        digit = (digit - digit % 10) / 10
    number = (number - number % 10) / 10

if card_type == 34 or card_type == 37:
    print("AMEX")
elif card_type >= 51 and card_type <= 55:
    print("MASTERCARD")
elif card_type >= 40 and card_type <= 49:
    print("VISA")
else:
    print("INVALID")

File: pset1/credit/credit.c
//This is poor implemenation
#include <stdio.h>
#include <cs50.h>

int main (void)
{
    long  every_second = 0, every = 0, leftover, every_second_sum = 0;
    long number = 0;
    int card_type = 0;
    do
    {
        number = get_long_long("Number:");
    }
    while (number <= 0);
    leftover = number;
    do                                          //this is a mess
    {
        if (leftover > 9 && leftover < 100)         //getting card type
        {
            card_type = leftover;
        }
        every = every + leftover % 10;
        leftover = (leftover - leftover % 10) / 10;
        every_second = (leftover % 10) * 2;
        if (every_second >= 10)
        {
            do
            {
                every_second_sum = every_second_sum + every_second % 10;
                every_second = (every_second - every_second % 10) / 10;
            }
            while (every_second > 0);

        }
        else
        {
            every_second_sum = every_second_sum + every_second % 10;
        }

        if (leftover > 9 && leftover < 100) //getting card type
        {
            card_type = leftover;
        }

        leftover = (leftover - leftover % 10) / 10;


    }
    while (leftover > 0);

//checking card type

    if (card_type == 34 || card_type == 37)
    {
        printf("AMEX\n");

    }
    else if (card_type >= 51 && card_type <= 55)
    {
        printf("MASTERCARD\n");

    }
    else if (card_type >= 40 && card_type <= 49)
    {
        printf ("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }

}

